plugins/catalog-import/src/util/useGithubRepos.ts

```diff
@@ -15,22 +15,29 @@
  */
 
 import * as YAML from 'yaml';
-import { useApi } from '@backstage/core';
+import { useApi, configApiRef } from '@backstage/core';
 import { catalogImportApiRef } from '../api/CatalogImportApi';
 import { ConfigSpec } from '../components/ImportComponentPage';
+import { readGitHubIntegrationConfigs } from '@backstage/integration';
 
 export function useGithubRepos() {
   const api = useApi(catalogImportApiRef);
+  const config = useApi(configApiRef);
 
   const submitPrToRepo = async (selectedRepo: ConfigSpec) => {
     const [ownerName, repoName] = selectedRepo.location.split('/').slice(-2);
+    const configs = readGitHubIntegrationConfigs(
+      config.getOptionalConfigArray('integrations.github') ?? [],
+    );
+    const githubIntegrationConfig = configs[0];
```

Comment:
We actually have the location of the repo right? So can match on the hostname instead of always selecting the first one.

Reply:
Next question is, are these config values even exposed in the frontend? Haha, I haven't checked.

Reply:
@freben They worked for me! So i guess so!

Reply:
Resolved in the latest commit, used the following: 
```ts
const [hostname, ownerName, repoName] = selectedRepo.location.split('/').slice(-3);
...
const githubIntegrationConfig = configs.find(v => v.host === hostname);
```

Reply:
Other parts of the repo is gathering around `git-url-parse`, search around the repo. This would be something like

```ts
import parseGitUrl from 'git-url-parse';
  const {
    name: repoName,
    owner: ownerName,
    resource: host,
  } = parseGitUrl(url);
```

Reply:
Looks a LOT better to me to use that. Thought the split-splice was a little iffy, but hey, i'm the guest here so i'll conform to what i find. :)

Reply:
@freben Now i'm also using git-url-parse. Worked like a charm!